src/server/game/Handlers/AuctionHouseHandler.cpp

```diff
@@ -736,24 +707,63 @@ void WorldSession::HandleAuctionListItems(WorldPacket& recvData)
         sortOrder.push_back(std::move(sortInfo));
     }
 
+    // converting string that we try to find to lower case
+    std::wstring wsearchedname;
+    if (!Utf8toWStr(searchedname, wsearchedname))
+        return;
+
+    wstrToLower(wsearchedname);
+
+    Creature* creature = GetPlayer()->GetNPCIfCanInteractWith(guid, UNIT_NPC_FLAG_AUCTIONEER);
+    if (!creature)
+        return;
+
     // remove fake death
     if (_player->HasUnitState(UNIT_STATE_DIED))
-    {
         _player->RemoveAurasByType(SPELL_AURA_FEIGN_DEATH);
-    }
 
-    // pussywizard:
-    const Milliseconds delay = 2s;
-    const Milliseconds now = GameTime::GetGameTimeMS();
-    Milliseconds diff = GetMSTimeDiff(_lastAuctionListItemsMSTime, now);
-    if (diff > delay)
```

Comment:
I was thinking maybe we could add rate limiter for some auction house opcodes somewhere [here](https://github.com/azerothcore/azerothcore-wotlk/blob/47580c0fab3d704ca5b2e4618984712e52e38950/src/server/game/Server/WorldSession.cpp#L1370). What do you think? Maybe we can address it in a separate PR if needed.

Reply:
The auction list opcodes definitely should be throttled, in fact there's even a blizzlike client side delay that we add to the list packets (now defined as AUCTION_SEARCH_DELAY) but that's obviously only on the client side. Personally, I've seen the antidos system and I can't say I'm a huge fan of how it works for a variety of reasons that would take too long to elaborate on, but I suppose it's the best we have for now. It could still use some refinement if we had a better system but as a temporary measure I've added them https://github.com/azerothcore/azerothcore-wotlk/pull/20830/commits/2a51d5629f534e3c0d4e8145dff9853bf0f1d2ab

CMSG_AUCTION_LIST_ITEMS is still lacking the blizzlike throttle for "getAll" requests (something around ~15 minutes if I recall correctly?). I don't believe it to be too important at the moment but it should also be implemented at some point. (Note: If someone other than me eventually implements it, there's a "gotcha" that CMSG_AUCTION_LIST_ITEMS is currently handled as PROCESS_THREADSAFE, any global containers used in the handler must be locked or the opcode moved to PROCESS_THREADUNSAFE)

Reply:
Yeah, agree. Thanks for addressing the comments!